MiniApp/Classes/MiniAppAdmobDisplayer.swift

```diff
@@ -0,0 +1,89 @@
+import Foundation
+import GoogleMobileAds
+
+/// Made to be a singleton
+/// It has to be a single instance to avoid multiple instances of Admob SDK
+class MiniAppAdmobDisplayer: NSObject, MiniAppAdDisplayProtocol {
+	var interstitial: GADInterstitial!
+	var rewarded: GADRewardedAd?
+	var onInterstitialClosed: (() -> Void)?
+	var onRewardedClosed: ((MiniAppReward?) -> Void)?
+	var lastReward: MiniAppReward?
+
+	override init() {
+		super.init()
+
+		GADMobileAds.sharedInstance().start(completionHandler: { [weak self] _ in
+			self?.createAndLoadInterstitial()
+			self?.createAndLoadRewarded()
+		})
+
+	}
+
+	private func createAndLoadRewarded() {
+		//TEST: Google's test id from tutorial
+		rewarded = GADRewardedAd(adUnitID: "ca-app-pub-3940256099942544/1712485313")
```

Comment:
Using a real id for testing can get your account suspended, so you should use Google's test ad unit IDs for testing: https://developers.google.com/admob/android/test-ads

Also, the ad unit IDs should be passed from the mini app (Javascript) side, so you don't need to have any IDs set on the native side. Our JS SDK Sample App is already set up to use test IDs, so you can use that for testing. See https://github.com/rakutentech/js-miniapp/blob/master/js-miniapp-sample/src/pages/ads.js#L103